### ISSUE TITLE
"Call to a member function getContentResourceHandle() on a non-object" when parsing large email files

### ISSUE THREAD

**Issue by aadilp** [zbateson/MailMimeParser#32]
I'm trying to get attachments from raw email data that is more than 9mb. The script runs just fine when the raw email is around 3-4mb. But generates this error when parsing a larger file:

`Call to a member function getContentResourceHandle() on a non-object`

Any idea what I might be doing wrong?

**Comment by aadilp** [comment]
Alright. So I tried `getAttachmentCount()` and it returns 0, even though the email has two attachments. And that's why the reference for the attachment object isn't created, resulting in the above error. Is there a fix for this?

**Comment by zbateson** [comment]
Hi @aadilp 

Can't think what that could be offhand but I'm wondering if it might be related to #31 as well.  What version of MMP are you using and would it be possible to send the message to me if it's not sensitive (zbateson at gmail)?  Either that or a description of the email for me to run tests:

1. Mime type of message and all parts (please include relationships if it's more than one level deep)
2. Size of each part since you suspect that may be the issue
3. Content-transfer-encoding of each part

I've tried some pretty large emails and haven't had any issues but I'm wondering if I may have introduced a bug in the latest version.

Thank you :)

**Comment by aadilp** [comment]
I tried parsing the same email on my local machine (using xampp on windows), and it worked. So the problem occurs on the production server, which is an EC2 instance running Ubuntu 14.04. I cleared the `/tmp` folder and tried parsing the email again, and this time it returned the attachment count as 1 (it's supposed to be 2). I saved the attachment to a file and it returned only about 742kb of the data (the original file is actually 2.15mb). It seems like the problem is with the server configuration, however `fread($file, filesize($filepath))` returns all the contents of the email. So I'm still confused about the root of the problem.
